addons/mail/static/src/discuss/discuss.xml

```diff
@@ -10,9 +10,21 @@
             <div class="px-3 bg-100 border-bottom d-flex flex-shrink-0" style="height:42px;">
                 <ThreadIcon className="'me-2 align-self-center'" thread="thread"/>
                 <div class="d-flex flex-grow-1 align-self-center align-items-center">
-                    <div class="lead flex-shrink-0 px-1 border text-truncate fw-bold border-light" t-att-title="thread.name" >
-                        <t t-esc="thread.name"/>
-                    </div>
+                    <t t-if="!state.isEditingThreadName">
+                        <div class="lead flex-shrink-0 px-1 border text-truncate fw-bold" t-att-title="thread.name" t-on-click="onClickThreadName" t-on-mouseenter="onMouseEnterThreadName" t-on-mouseleave="onMouseLeaveThreadName" t-attf-class="{{ state.isMouseOverThreadName ? 'o-threadNameEditable bg-white' : 'border-light' }}">
```

Comment:
I have a suggestion: maybe we could always render an input in all cases, so we don't need to do all that complicated switch work. Simply style the input to make it look like a div, and then it is there ready to be used

Reply:
yes, but the problem I met was I could not find the correct time to provide the `thread.name` to `pendingName`, which is used for the t-model. The reason to do the switch is to manually trigger the `pendingName = thread.name`. Any solution for the time to do the trigger?

Reply:
oh yeah, I see.  There is a way to observe the discuss threadId variable and react to it. See an example here: https://github.com/odoo-dev/odoo/commit/a2cb93b2a83c1d4efdab919d781e5904f7e85a75 I thought several times about introducing this in `web`. What do you think?

Reply:
Thanks! I think this is pretty handy for me. Otherwise, I have to check it every time the page is rendered. It would be nice if this observer can be introduced in `web`